tech-notes.md

```diff
@@ -0,0 +1,50 @@
+## Tech Notes
+
+There are 6 canvases on the page:
+
+* `imageCanvas` is the only visible canvas. It's where the composite image is rendered and is downloaded when the user is done.
+* `tempCanvas` holds path data when drawing paths.
+* `holderCanvas` takes a snapshot of the `imageCanvas` as soon as a new path starts.
+* `rotationCanvas`
+* `blurredCanvas` holds a blurred version of the current image, re-blurred on each mouseup.
+* `offscreenCanvas`, which is used to render the custom cursor.
+
+### Opening an image
+
+When you open an image:
+
+1. All canvases are resized to match the image's dimensions, scaled so that no side is 2500px or more.
+1. The image is drawn on `imageCanvas` and `rotationCanvas`.
+1. The brush size and blur radius adjust themselves to the size of the image - the larger the image, the larger the brush and blur.
+1. The cursor is rendered offscreen and saved to a data URL in CSS.
+
+### Drawing a path
+
+When you click or tap on the canvas:
+
+1. **On mouse down/tap start:**
+    1. The current `imageCanvas` image is copied to the `holderCanvas`.
+    1. The `tempCanvas` is cleared. 
+    1. The mouse/touch position is saved as `lastPos`.
+1. **On mouse/touch move:**
+    1. A circle is drawn at each point to the `imageCanvas` and `tempCanvas`.
```

Comment:
And a rectangle is interpolated between the new position and `lastPos`

Reply:
Noted and added, thanks!